sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -588,7 +588,7 @@ hintStatement
     ;
 
 fromClause
-    : FROM relation (',' relation)* lateralView* pivotClause?
+    : FROM relation (',' lateralClause)* (',' relation)* lateralView* pivotClause?
```

Comment:
Interesting. But what outer references we can use? It's the leftmost query...

Reply:
Yea, it seems PosgreSQL accepts simple cases without outer refs.

Reply:
I think it was mentioned in the [manual](https://www.postgresql.org/docs/current/queries-table-expressions.html#QUERIES-LATERAL):

> A LATERAL item can appear at top level in the FROM list, or within a JOIN tree. In the latter case it can also refer to any items that are on the left-hand side of a JOIN that it is on the right-hand side of.